addons/mail/static/tests/message/message_tests.js

```diff
@@ -672,6 +673,44 @@ QUnit.test("should not be able to reply to temporary/transient messages", async
     await contains(".o-mail-Message [title='Reply']", { count: 0 });
 });
 
+QUnit.test("squashed transient message should not have date in the sidebar", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "Channel 1" });
+    pyEnv["mail.message"].create([
+        {
+            body: "Hello world 1",
+            model: "discuss.channel",
+            res_id: channelId,
+            date: "2024-02-26 10:00:00",
+        },
+        {
+            body: "Hello world 2",
+            model: "discuss.channel",
+            res_id: channelId,
+            date: "2024-02-26 10:00:00",
+        },
+    ]);
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click(".o-mail-Message.o-squashed");
+    await contains(".o-mail-Message.o-squashed .o-mail-Message-sidebar .o-mail-Message-date", {
+        text: deserializeDateTime("2024-02-26 10:00:00").toLocaleString(DateTime.TIME_24_SIMPLE),
+    });
+    await insertText(".o-mail-Composer-input", "/who");
+    await click(".o-mail-Composer-send:enabled");
+    await contains(".o-mail-Message", { count: 3 });
+    await insertText(".o-mail-Composer-input", "/who");
+    await click(".o-mail-Composer-send:enabled");
+    await click(":nth-child(2 of .o-mail-Message.o-squashed");
+    await nextTick();
+    await contains(
+        ":nth-child(2 of .o-mail-Message.o-squashed) .o-mail-Message-sidebar .o-mail-Message-date",
+        {
+            count: 0,
+        }
+    );
```

Comment:
Better assert no `text: "10:00"` like how the end-user would check the non-presence of feature. inexistence of classname `.o-mail-Message-date` doesn't mean anything to end-user